### PR TITLE
Add the option of giving device series name while performing the task…

### PR DESCRIPTION
**Pull request by Abhishek-121**
… of SWIM Image Distribution/Activation along with giving the name of site, device family and device role, check with regex for device series name, update examples and docs as well.

In this commit -

— Add the option of giving device series name while performing the task of SWIM Image Distribution/Activation along with giving the name of site, device family and device role.

— Check with regex for device series name that will check the string given in the playbook’s input.

— Write and update documentation and examples for same and in the playbook as well.

### CODE REVIEW

File: plugins/modules/swim_intent.py [hunk 1]
```diff
@@ -730,15 +741,25 @@ def get_device_uuids(self, site_name, device_family, device_role):
 
         device_uuid_list = []
         if not site_name:
+            self.log("Site name is not given so cannot fetch the devices associated to site", "INFO")
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:744]: updated.

File: plugins/modules/swim_intent.py [hunk 2]
```diff
@@ -730,15 +741,25 @@ def get_device_uuids(self, site_name, device_family, device_role):
...
             return device_uuid_list
 
         (site_exists, site_id) = self.site_exists(site_name)
         if not site_exists:
+            self.log("""Given site '{0}' is not present in Cisco Catalyst Center so cannot fetch the devices associated
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:749]: updated.

File: plugins/modules/swim_intent.py [hunk 3]
```diff
@@ -730,15 +741,25 @@ def get_device_uuids(self, site_name, device_family, device_role):
...
+                      to site""".format(site_name), "INFO")
             return device_uuid_list
 
+        if device_series_name:
+            if device_series_name.startswith(".*") and device_series_name.endswith(".*"):
+                self.log("Device series name '{0}' already present in the regex format".format(device_series_name), "INFO")
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:755]: updated.

File: plugins/modules/swim_intent.py [hunk 4]
```diff
@@ -752,8 +773,12 @@ def get_device_uuids(self, site_name, device_family, device_role):
         if len(response) > 0:
             for item in response:
                 if item["reachabilityStatus"] != "Reachable":
+                    self.log("""Reachability status of device '{0}' is '{1}' so cannot add it for distribution/activation
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:776]: added.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**